python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -421,27 +420,12 @@ def do_evaluate(
             return Column(plc_column, dtype=self.dtype)
         elif self.name is StringFunction.Name.JsonDecode:
             plc_column = self.children[0].evaluate(df, context=context).obj
-            # Once https://github.com/rapidsai/cudf/issues/19338 is implemented,
-            # we can use do this conversion on device.
-            buff = io.StringIO(
-                plc.strings.combine.join_strings(
-                    plc_column,
-                    plc.Scalar.from_py("\n", plc_column.type()),
-                    plc.Scalar.from_py("NULL", plc_column.type()),
-                )
-                .to_scalar()
-                .to_py()
-            )
-            source = plc.io.types.SourceInfo([buff])
-            options = (
-                plc.io.json.JsonReaderOptions.builder(source)
-                .lines(val=True)
-                .dtypes(_dtypes_for_json_decode(self.dtype))
-                .compression(plc.io.types.CompressionType.NONE)
-                .recovery_mode(plc.io.types.JSONRecoveryMode.RECOVER_WITH_NULL)
-                .build()
+            plc_table_with_metadata = plc.io.json.read_json_from_string_column(
+                plc_column,
+                plc.Scalar.from_py("\n", plc_column.type()),
+                plc.Scalar.from_py("NULL", plc_column.type()),
```

Comment:
the type must be STRING anyway so...
```suggestion
                plc.Scalar.from_py("\n"),
                plc.Scalar.from_py("NULL"),
```